enterprise_access/apps/api/v1/views/customer_billing.py

```diff
@@ -181,37 +199,135 @@ def create_checkout_session(self, request, *args, **kwargs):
-        except requests.exceptions.HTTPError:
-            return Response(None, status=status.HTTP_404_NOT_FOUND)
-
-        # Next, create a stripe customer portal session.
-        customer_portal_session = stripe.billing_portal.Session.create(
-            customer=enterprise_customer_data['stripe_customer_id'],
-            return_url=f"https://portal.edx.org/{enterprise_customer_data['slug']}",
+            customer_portal_session = stripe.billing_portal.Session.create(
+                customer=stripe_customer_id,
+                return_url="https://enterprise-checkout.stage.edx.org/billing-details/success",
+            )
+        except stripe.error.StripeError as e:
+            # Generic catch-all for other Stripe errors
+            logger.exception(
+                f"StripeError creating billing portal session for CheckoutIntent {checkout_intent}: {e}",
+            )
+            return Response(customer_portal_session, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+        except Exception as e:
+            # Any other unexpected error
+            logger.exception(
+                f"General exception creating billing portal session for CheckoutIntent {checkout_intent}: {e}",
+            )
+            return Response(customer_portal_session, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+        # TODO: pull out session fields actually needed, and structure a response.
+        return Response(
+            customer_portal_session,
+            status=status.HTTP_200_OK,
+            content_type='application/json',
         )
+    @extend_schema(
+        tags=[CUSTOMER_BILLING_API_TAG],
+        summary='Create a new Customer Portal Session from the enterprise checkout MFE.',
+    )
+    @action(
+        detail=True,
+        methods=['get'],
+        url_path='create-checkout-portal-session',
+    )
+    def create_checkout_portal_session(self, request, pk=None):
+        """
+        Create a new Customer Portal Session for the enterprise checkout MFE.  Response dict contains "url" key
+        that should be attached to a button that the customer clicks.
+
+        Response structure defined here: https://docs.stripe.com/api/customer_portal/sessions/create
+        """
+        customer_portal_session = None
+        origin_url = request.META.get("HTTP_ORIGIN")
+        checkout_intent = CheckoutIntent.objects.filter(pk=int(pk)).first()
+
+        if not checkout_intent:
+            logger.error(f"No checkout intent for id, for requesting user {request.user.id}")
+            return Response(customer_portal_session, status=status.HTTP_404_NOT_FOUND)
+
+        stripe_customer_id = checkout_intent.stripe_customer_id
+        if not stripe_customer_id:
+            logger.error(f"No stripe customer id associated to CheckoutIntent {checkout_intent}")
+            return Response(customer_portal_session, status=status.HTTP_404_NOT_FOUND)
+
+        if not checkout_intent:
```

Comment:
There a lot of code that is common between the two new endpoints.  I'd recommend making this DRY and consolidating as much common code into a single function, possibly an internal python API function stored in the customer_billing domain (e.g. `enterprise_access/apps/customer_billing/api.py`)